src/fsharp/FSharp.Core/prim-types.fs

```diff
@@ -2218,6 +2220,7 @@ namespace Microsoft.FSharp.Core
                 | ty when ty.Equals(typeof<float32>)    -> unboxPrim (box Float32Comparer)
                 | ty when ty.Equals(typeof<decimal>)    -> unboxPrim (box DecimalComparer)
                 | ty when ty.Equals(typeof<string>)     -> unboxPrim (box StringComparer)
+                | ty when ty.Equals(typeof<bool>)     -> unboxPrim (box BoolComparer)
```

Comment:
nit: that code was vertically aligned, we should keep it aligned.

I'm baffled there are specialized cases for many types but not for bool!